### PR TITLE
Assignment week 3

### PR DESCRIPTION
**Pull request by boaloysius**

### CODE REVIEW

File: Week-3/q1.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+function time(fn) {
+  console.time();
+  fn();
+  console.timeEnd();
+}
+
+const memoizeAdd = memoize(add);
+
+time(() => console.log(memoizeAdd(38, 30, 100, 200, 5000)));
+time(() => console.log(memoizeAdd(38, 30, 100, 200, 5000)));
```
Comment (anilrao316) [inline Week-3/q1.js:26]: what if the order of parameters are changed? will it work the same way ?

File: Week-3/q4.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+function createStack() {
+  let items = [];
```
Comment (anilrao316) [inline Week-3/q4.js]: can be const

File: Week-3/q1.js (hunk 1 above)
Comment (boaloysius) [reply Week-3/q1.js:26]: We can sort the parameters and use as key for cache

Comment (anilrao316) [reply Week-3/q1.js:26]: right

**Pull request merged**